forem/forem
# Default to Imgproxy on Forem-cloud

## What type of PR is this? (check all applicable)

## Description
This feature only applies to Forem-cloud's apps, ie https://forem.dev. It will default all image resizing to Imgproxy instead of Cloudinary.

## Related Tickets & Documents
https://github.com/orgs/forem/projects/11#card-44712245
## QA Instructions, Screenshots, Recordings
heads-up: if you re-save anything during this process, it may break some image links. Simply resave afterward again to fix it, or you can nuke your DB.

1. You will need to install imgproxy locally. You can do so with `brew install imgproxy`
2. You need to generate a hex-encoded key/salt pair for this to work. In a terminal, rung `echo $(xxd -g 2 -l 64 -p /dev/random | tr -d '\n')` twice and copy those values to your `.env` in the next step
3. In your `.env` set the following
```
export IMGPROXY_ENDPOINT= 'http://localhost:8080'
export IMGPROXY_KEY='1b1c9aae804e070b0864f2547fba7ce8ff31bf7..........'
export IMGPROXY_SALT='8c6d449d4fc2cada5bab538826cae709d2ade9f9286af8b83573346bfb7.........'
```
4. In a separate terminal tab, start imgproxy server by running `IMGPROXY_KEY='your key' IMGPROXY_SALT='your salt' imgproxy`
5. Start rails server normally.
6. Load up development and please be sure to sign in. You should see all the profile image loaded correctly, especially your own profile image in the navbar. When you open up any user's profile image, the URL should look similar to this structure`http://localhost:8080/4R6Fv5Jy_-0PiD-53tMGX2sGPde78X3IFWNl7ETO89Q/s:320:320/...`.

## Added tests?
- [x] updated the existing test.

## Added to documentation?
- [x] no, will add in a different PR

## [optional] Are there any post-deployment tasks we need to perform?
n/a for dev.to. Forem-cloud apps will need to run the included rake task to resave all articles and comments.
